invokeai/app/services/events.py

```diff
@@ -46,6 +47,7 @@ def emit_generator_progress(
                 source_node_id=source_node_id,
                 progress_image=progress_image.dict() if progress_image is not None else None,
                 step=step,
+                order=order,
```

Comment:
Can you write the formula for this here in the PR please? for reference when updating frontendd

Reply:
To be 100% accurate it's:
```python
if order == 2:
    step = (step + 1) // 2
    total_steps = (total_steps + 1) // 2
```
For example Heun 20 steps - will report total_steps=39
In theory there could be schedulers with higher order, but there no such schedulers in diffusers currently.

Reply:
To generalize:
```py
step = (step + 1) // order
total_steps = (total_steps + 1) // order
```

Or is `2` unrelated to `order`?

Are the valid `order`s currently `1 | 2`?

Reply:
Yes, currently there exists only 1 and 2(heun, kdpm2, dpmpp_sde).
I not sure how it's correct to generalize:
```python
if order > 1:
    step = (step + 1) // order
    total_steps = (total_steps + 1) // order
```
or
```python
if order > 1:
    step = (step + order - 1) // order
    total_steps = (total_steps + order - 1) // order
```
So... that's why I wrote only about order=2)

Reply:
We shouldn't have to reverse-engineer this; the thing emitting the progress events has access to the complete list of timesteps it's planning to run.

Reply:
I send raw values because we can show progress more smooth, than when we will skip every second unet run on order=2 schedulers.
Sended before code with `(v + 1) // 2` already output correct values.
Or if you about schedulers internals - I read them, but here no comments why it's done this way.
I understand why it repeat each timestep twice(because they do different calculations on each step), but why for example first timestep not repeated too - there no comments.